File: benchmarks/js/call/fibonacci.js
// Naive recursive Fibonacci benchmark

var print = typeof console !== 'undefined' ? console.log.bind(console) : globalThis.print;
var DEFAULT = 35;
var N = parseInt(typeof scriptArgs !== 'undefined' ? scriptArgs[1] : (typeof process !== 'undefined' ? process.argv[2] : '0')) || DEFAULT;

function fib(n) {
    if (n < 2) {
        return n;
    }
    return fib(n - 1) + fib(n - 2);
}

var result = fib(N);
print("fib(" + N + ") = " + result);

File: benchmarks/lua/call/fibonacci.lua
-- Naive recursive Fibonacci benchmark

local N = tonumber(arg and arg[1]) or 38
collectgarbage("collect")

local function fib(n)
    if n < 2 then
        return n
    end
    return fib(n - 1) + fib(n - 2)
end

local result = fib(N)
print("fib(" .. N .. ") = " .. result)
